flink-connector-bigquery/src/main/java/com/google/cloud/flink/bigquery/table/restrictions/BigQueryPartition.java

```diff
@@ -69,16 +274,17 @@ public static List<String> partitionValuesFromIdAndDataType(
                 // lets first check that all the partition ids have the same length
                 String firstId = partitionIds.get(0);
                 Preconditions.checkState(
-                        partitionIds.stream().allMatch(pid -> pid.length() == firstId.length()),
+                        partitionIds.stream()
+                                .allMatch(
+                                        pid ->
+                                                (pid.length() == firstId.length())
+                                                        && StringUtils.isNumeric(pid)),
                         "Some elements in the partition id list have a different length: "
                                 + partitionIds.toString());
                 switch (firstId.length()) {
                     case 4:
                         // we have yearly partitions
-                        partitionValues.addAll(
-                                partitionIds.stream()
-                                        .map(id -> String.format("'%s'", id))
-                                        .collect(Collectors.toList()));
+                        partitionValues.addAll(partitionIds.stream().collect(Collectors.toList()));
```

Comment:
nit: looks like we can just directly them. `partitionValues.addAll(partitionIds);`

Reply:
Changed